src/Func.h

```diff
@@ -136,6 +165,18 @@ struct CallInfo {
 	const val_list* args;
 };
 
+// Struct that collects the arguments for a Func.
+// Used for BroFuncs with closures.
+struct function_ingredients
```

Comment:
I haven't traced this through: is the reference counting done where necessary when storing pieces in these ingredients? Alternatively, the struct could maintain ref counts itself for the duration of an instance's lifetime.